chain-test/src/test/scala/org/bitcoins/chain/models/BlockHeaderDAOTest.scala

```diff
@@ -0,0 +1,201 @@
+package org.bitcoins.chain.models
+
+import org.bitcoins.chain.db.ChainDbManagement
+import org.bitcoins.chain.util.ChainUnitTest
+import org.bitcoins.core.crypto.{DoubleSha256Digest, ECPrivateKey}
+import org.bitcoins.core.number.{Int32, UInt32}
+import org.bitcoins.core.protocol.blockchain.BlockHeader
+import org.bitcoins.testkit.chain.ChainTestUtil
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+/**
+  * Created by chris on 9/8/16.
+  */
+class BlockHeaderDAOTest extends ChainUnitTest {
+  implicit val ec: ExecutionContext =
+    scala.concurrent.ExecutionContext.Implicits.global
+
+  val blockHeaderDAO = BlockHeaderDAO(chainParams =
+                                        ChainTestUtil.regTestChainParams,
+                                      dbConfig = dbConfig)
+  before {
+    //Awaits need to be used to make sure this is fully executed before the next test case starts
+    //TODO: Figure out a way to make this asynchronous
+    Await.result(ChainDbManagement.createHeaderTable(dbConfig), timeout)
+    Await.result(blockHeaderDAO.create(genesisHeader), timeout)
+  }
```

Comment:
This could probably be resolved (along with the `after` and some redundancy in tests) by using fixtures